BlueprintUI/Sources/Layout/Builder/StackChild.swift

```diff
@@ -0,0 +1,107 @@
+import UIKit
+
+
+/// `StackChild` is a wrapper which holds an element along with  its StackLayout traits and Keys.
+/// This struct is particularly useful when working with `@ElementBuilder<StackChild>`.
+/// By default, elements inside of `ElementBuilder<StackChild>` will be implicitly converted to a StackChild
+/// with a nil key and the default `StackLayout.Traits` initializer. But when given a `StackChild` via
+/// `Element.stackChild(...)` modifier or initialized directly, `@ElementBuilder<StackChild>`
+/// pull out the given traits and key and then apply those to the stack
+public struct StackChild {
+    public let element: Element
+    public let traits: StackLayout.Traits
+    public let key: AnyHashable?
+
+    public enum Priority {
+        case fixed
+        case flexible
+        case grows
+        case shrinks
```

Comment:
I worry that `grows` and `shrinks` are more likely to be used incorrectly than to be useful.

Reply:
IMO it's more useful / more clear than having people pass `0` and `1` to the individual params. What about more verbose names?

`growsOnUnderflow`
`shrinksOnOverflow`

Reply:
I like `growsOnUnderflow` and `shrinksOnOverflow` but then I would expect something like `growsAndShrinks` instead of `flexible` for consistent semantics. Otherwise I wouldn't immediate catch that `flexible` is a combination of both `grows` and `shrinks`.

Reply:
IMO they're special enough that I think it's fine to have slightly different naming than the two common use cases.